api/plugins/ginfo/ginfo.py

```diff
@@ -0,0 +1,153 @@
+
+            @param group: UID of the ginfo group for which the marker should be posted
+            @param marker: UID of the marker that should be posted
+        """
+        return "groupAccessTokenVerifier/" + group + "/" + marker
+
+    TRANSFORMATION_KX = 81.48592277749488
+    TRANSFORMATION_KY = -81.9610812970396
+    TRANSFORMATION_DX = 6275917.802625263
+    TRANSFORMATION_DY = 11160459.85122114
+
+    def linear_transform(self, x, y):
+        """
+            Applies a linear transform to the point x/y
+        """
+        return (
+            x * self.TRANSFORMATION_KX + self.TRANSFORMATION_DX,
+            y * self.TRANSFORMATION_KY + self.TRANSFORMATION_DY
+        )
+
+    def convert_position(self, x, y):
+        """
+          Converts the given position from x/y to lat/lng coordinates
+          @param x, y: x/y Position of in the game
+          @return: lat/lng for the same position on the ginfo map
+        """
+        (x, y) = self.linear_transform(x, y)
+        return SphericalMercator.unproject(x, y)
+
+    DEFAULT_MARKER_COLOR = "#1D8BF1"
+    DEFAULT_MARKER_TYPE = 1
+    DEFAULT_MARKER_SKIN = 0
+
+    def update_position(self, character, group, access_token):
+        """
+          Posts the position of this character to Firebase
+          @param character:
+                    The character for which the position should be updated
+          @param group:
+                    The group in which the marker should be postet
+          @param access_token:
+                    Access token for the ginfo group
+        """
+        (lat, lng) = self.convert_position(
+            float(character.x), float(character.y))
+
+        ginfo_character, created = GinfoCharacter.objects.get_or_create(
+            character_id=character.id)
+        if created:
+            ginfo_character.ginfo_marker_uid = self.uid_generator.next_id()
+            ginfo_character.save()
+
+        data = {
+            self.marker_path(group, ginfo_character.ginfo_marker_uid): {
+                "name": character.name,
+                "updatedAt": {
+                    # This tells firebase to use its internal timestamp
+                    ".sv": "timestamp"
+                },
+                "createdAt": {
+                    # This tells firebase to use its internal timestamp
+                    ".sv": "timestamp"
+                },
+                "color": self.DEFAULT_MARKER_COLOR,
+                "creatorId": self.GINFO_USER_UID,
+                "position": {
+                    "lat": lat,
+                    "lng": lng
+                },
+                "type": self.DEFAULT_MARKER_TYPE,
+                "skin": self.DEFAULT_MARKER_SKIN,
+            },
+            self.access_token_path(group, ginfo_character.ginfo_marker_uid): access_token
+        }
+
+        response = requests.patch(
+            url=self.API_URL + '/.json',
+            json=data
+        )
+        json_data = response.json()
```

Comment:
this can crash if firebase returns no body, after line 117 you should maybe check for no body, or a 500 error and print a different error. I'll add newrelic error reporting later to handle this better.